layers/+tools/ansible/config.el

```diff
@@ -19,4 +19,4 @@ If non-nil then encrypted files are automatically decrypted when opened and
 ;; detect filenames compatible with Ansible's recommended layout.
 ;; http://docs.ansible.com/playbooks_best_practices.html#directory-layout
 (setq spacemacs--ansible-filename-re
-      ".*\\(\\(main\\|site\\|encrypted\\|roles/.+\\)\.yml\\|group_vars/.+\\|host_vars/.+\\)")
+      ".*\\(\\(main\\|site\\|encrypted\\|roles/.+\\).yml\\|group_vars/.+\\|host_vars/.+\\)")
```

Comment:
I think the regexp is wrong as it is, currently it would match `main-yml` which is wrong. To fix it the `.` should be escaped with double `\\`. 

You can test it with below code:
```
;; Without \
(string-match ".*\\(\\(main\\|site\\|encrypted\\|roles/.+\\).yml\\|group_vars/.+\\|host_vars/.+\\)" "main-yml") => Matches
(string-match ".*\\(\\(main\\|site\\|encrypted\\|roles/.+\\).yml\\|group_vars/.+\\|host_vars/.+\\)" "main.yml") => Matches
(string-match ".*\\(\\(main\\|site\\|encrypted\\|roles/.+\\).yml\\|group_vars/.+\\|host_vars/.+\\)" "main..yml") => Matches


;; With \\
(string-match ".*\\(\\(main\\|site\\|encrypted\\|roles/.+\\)\\.yml\\|group_vars/.+\\|host_vars/.+\\)" "main-yml") => No match
(string-match ".*\\(\\(main\\|site\\|encrypted\\|roles/.+\\)\\.yml\\|group_vars/.+\\|host_vars/.+\\)" "main.yml") => Matches
(string-match ".*\\(\\(main\\|site\\|encrypted\\|roles/.+\\)\\.yml\\|group_vars/.+\\|host_vars/.+\\)" "main..yml") => Matches


;; Original
(string-match ".*\\(\\(main\\|site\\|encrypted\\|roles/.+\\)\.yml\\|group_vars/.+\\|host_vars/.+\\)" "main-yml") => Matches
(string-match ".*\\(\\(main\\|site\\|encrypted\\|roles/.+\\)\.yml\\|group_vars/.+\\|host_vars/.+\\)" "main.yml") => Matches
(string-match ".*\\(\\(main\\|site\\|encrypted\\|roles/.+\\)\.yml\\|group_vars/.+\\|host_vars/.+\\)" "main..yml") => Matches
```

Reply:
Your right, thanks for the correction.

Was the result `=> Matches` for the `main..yml` tests a typo?

Because all three `main..yml` tests return `nil` (no match) for me. And it seems to be expected since:
`The dot matches a single character`
source: https://www.regular-expressions.info/dot.html

Reply:
```elisp
(dolist (re '(;; Without \
              (".*\\(\\(main\\|site\\|encrypted\\|roles/.+\\).yml\\|group_vars/.+\\|host_vars/.+\\)" "main-yml")
              (".*\\(\\(main\\|site\\|encrypted\\|roles/.+\\).yml\\|group_vars/.+\\|host_vars/.+\\)" "main.yml")
              (".*\\(\\(main\\|site\\|encrypted\\|roles/.+\\).yml\\|group_vars/.+\\|host_vars/.+\\)" "main..yml")
              ;; With \\
              (".*\\(\\(main\\|site\\|encrypted\\|roles/.+\\)\\.yml\\|group_vars/.+\\|host_vars/.+\\)" "main-yml")
              (".*\\(\\(main\\|site\\|encrypted\\|roles/.+\\)\\.yml\\|group_vars/.+\\|host_vars/.+\\)" "main.yml")
              (".*\\(\\(main\\|site\\|encrypted\\|roles/.+\\)\\.yml\\|group_vars/.+\\|host_vars/.+\\)" "main..yml")
              ;; Original
              (".*\\(\\(main\\|site\\|encrypted\\|roles/.+\\)\.yml\\|group_vars/.+\\|host_vars/.+\\)" "main-yml")
              (".*\\(\\(main\\|site\\|encrypted\\|roles/.+\\)\.yml\\|group_vars/.+\\|host_vars/.+\\)" "main.yml")
              (".*\\(\\(main\\|site\\|encrypted\\|roles/.+\\)\.yml\\|group_vars/.+\\|host_vars/.+\\)" "main..yml")))
  (message "%s %s => %s" (car re) (cadr re)
           (string-match (car re) (cadr re))))
```
outputs:
```
.*\(\(main\|site\|encrypted\|roles/.+\).yml\|group_vars/.+\|host_vars/.+\) main-yml => 0
.*\(\(main\|site\|encrypted\|roles/.+\).yml\|group_vars/.+\|host_vars/.+\) main.yml => 0
.*\(\(main\|site\|encrypted\|roles/.+\).yml\|group_vars/.+\|host_vars/.+\) main..yml => nil
.*\(\(main\|site\|encrypted\|roles/.+\)\.yml\|group_vars/.+\|host_vars/.+\) main-yml => nil
.*\(\(main\|site\|encrypted\|roles/.+\)\.yml\|group_vars/.+\|host_vars/.+\) main.yml => 0
.*\(\(main\|site\|encrypted\|roles/.+\)\.yml\|group_vars/.+\|host_vars/.+\) main..yml => nil
.*\(\(main\|site\|encrypted\|roles/.+\).yml\|group_vars/.+\|host_vars/.+\) main-yml => 0
.*\(\(main\|site\|encrypted\|roles/.+\).yml\|group_vars/.+\|host_vars/.+\) main.yml => 0
.*\(\(main\|site\|encrypted\|roles/.+\).yml\|group_vars/.+\|host_vars/.+\) main..yml => nil
```

Reply:
Hmm strange thought that he matched main..yml for me, now it does not, seems to be some strange issue with `, e e`. Anyway now main-yml is not longer matched which is great :).